enterprise/functions/src/main/java/io/crate/window/OffsetValueFunctions.java

```diff
@@ -36,20 +36,40 @@
 import java.util.List;
 
 /**
- * The lag function returns the evaluated value at a row that
- * precedes the current row by offset within its partition.
+ * The offset functions return the evaluated value at a row that
+ * precedes or follows the current row by offset within its partition.
  * <p>
  * Synopsis:
- *      lag(argument any [, offset integer [, default any ] ])
+ *      lag/lead(argument any [, offset integer [, default any ] ])
  */
-public class LagFunction implements WindowFunction {
+public class OffsetValueFunctions implements WindowFunction {
 
-    private static final String NAME = "lag";
 
+    private enum OffsetDirection {
```

Comment:
lgtm 👍 

Did you consider using just `IntBinaryOperator` for index calculation or it becomes a bit less readable?

Reply:
I think it's a bit clearer this way, but maybe it's just a personal preference.